### ISSUE TITLE
Drawing circuits with custom instructions

### ISSUE THREAD

**Issue by TimAlexisKoerner** [Qiskit/qiskit-terra#6575]
### Information

- **Qiskit Terra version**:0.17.3
- **Python version**:3.9.4
- **Operating system**:Windows

### What is the current behavior?
Drawings of circuits with custom instruction using output mode text, 
where the instructions are given bits in non-ascending order,
results in overflowing box representation.
![custominst_wrongorder](https://user-images.githubusercontent.com/83113669/121946390-2a44d380-cd55-11eb-8f84-4fec42a64f94.jpg)
Other output options do not support classical bits at all.

### Steps to reproduce the problem
```
from qiskit.circuit import Instruction, QuantumCircuit, QuantumRegister

class CustomInstruction(Instruction):
    def __init__(self):
        super().__init__("custom_I", 2, 2, [])
    def _define(self):
        qreg = QuantumRegister(2, "q")
        creg = QuantumRegister(2, "c")
        qc = QuantumCircuit(qreg, creg)
        self.defintion = qc

qc = QuantumCircuit(2,2)
qc.append(CustomInstruction(), [0,1], [0,1])
qc.append(CustomInstruction(), [0,1], [1,0])
qc.append(CustomInstruction(), [1,0], [0,1])
qc.append(CustomInstruction(), [1,0], [1,0])
qc.draw(output="text")
```

### What is the expected behavior?
All instructions should look like the first shown in the picture.

### Suggested solutions
Sort given bits for drawing (non inplace).

**Comment by TharrmashasthaPV** [comment]
Hi @TimAlexisKoerner . I believe this issue has been resolved in PR #6242 . You should be able to see the fix in Qiskit Terra version 0.18.

**Comment by TimAlexisKoerner** [comment]
Thank you for the notification, i will close this issue.
